### DISCUSSION TITLE
Parsing Additional Date Array Values

### DISCUSSION THREAD

**Discussion by HYPERTONE** [rtosholdings/riptable#320] [Ideas] [answered]
Currently we can parse Date arrays for month and year via the following:

```
>>> d = Date(['2000-02-29', '2018-12-25'])
>>> d.month
>>> d.year
```

However it would be fantastic if we could parse additional components, such as:
- Hour
- Minute
- Second
- Microsecond
- Nanosecond

I.e. Given the following `Date(['2018-12-25 12:45:30.123456000']`, parse all 'hour' components into a Fast Array and/or put into a new column.

**Comment by OrestZborowski-SIG** [comment, marked as answer]
There are additional date/time types available, such as `DateTimeNano`. Could those be useful here?

**Reply by HYPERTONE** [reply]
Yes! 

For anyone else who stumbles upon this.

Given the following:
`dtn = rt.DateTimeNano(['6/1/2022 12:00:03 AM'], from_tz='GMT', to_tz='GMT', format="%m/%d/%Y %H:%M:%S")`

We can output the hour via:
```
dtn.hour
>> FastArray([12.00083333])
```

Where 0.00083333 is the hourly conversion of 3 seconds from the timestamp.

If you only wish to preserve the hour, you can type cast via the following:
`dtn.hour.astype(int)`

~8,000,000 rows in ~3 seconds to process. Very cool indeed!
